src/transformers/generation/tf_logits_process.py

```diff
@@ -585,3 +586,137 @@ def _force_token(generation_idx):
+        #     last_was_timestamp = len(seq) >= 1 and seq[-1] >= self.timestamp_begin
+        #     penultimate_was_timestamp = len(seq) < 2 or seq[-2] >= self.timestamp_begin
+        #     if last_was_timestamp:
+        #          if penultimate_was_timestamp:  # has to be non-timestamp
+        #               scores[k, self.timestamp_begin :] = -float("inf")
+        #          else:  # cannot be normal text tokens
+        #               scores[k, : self.eos_token_id] = -float("inf")
+
+        def _update_slices(batch_idx_to_modify, idx0, idx1):
+            batch_idx_to_modify = tf.where(batch_idx_to_modify)
+            new_scores = tf.identity(scores)
+            if batch_idx_to_modify.shape[0] is not None:
+                if batch_idx_to_modify.shape[0] > 0:
+                    batches = batch_idx_to_modify.shape[0]
+                    mod = tf.tile(range(idx0, idx1), [batches])
+                    batch = tf.repeat(batch_idx_to_modify, (idx1 - idx0))
+                    final_idx = tf.stack((tf.cast(batch, mod.dtype), mod), axis=1)
+                    updates = tf.ones(((idx1 - idx0) * batches,), dtype=scores.dtype) * -float("inf")
+                    new_scores = tf.tensor_scatter_nd_update(new_scores, final_idx, updates)
+            return new_scores
+
+        max_idx = scores.shape[1]
+        indices_1 = last_was_timestamps & penultimate_was_timestamp
+        indices_2 = last_was_timestamps & tf.math.logical_not(penultimate_was_timestamp)
+
+        scores = _update_slices(indices_1, self.timestamp_begin, max_idx)
+        scores = _update_slices(indices_2, 0, self.eos_token_id)
+
+        # if input_ids.shape[1] == self.begin_index and self.max_initial_timestamp_index is not None:
+        #     last_allowed = self.timestamp_begin + self.max_initial_timestamp_index
+        #     scores[:, last_allowed + 1 :] = -float("inf")
+
+        scores = tf.cond(
+            tf.equal(cur_len, self.begin_index) and self.max_initial_timestamp_index is not None,
+            lambda: _update_slices(
+                tf.ones((scores.shape[0],), dtype=tf.bool),
+                self.timestamp_begin + self.max_initial_timestamp_index + 1,
+                max_idx,
+            ),
+            lambda: tf.identity(scores),
+        )
+
+        ####################################################################################################
+        # Still left to do: TODO this does not seem to properly update the scores
+        # logprobs = torch.nn.functional.log_softmax(scores.float(), dim=-1)
+        # for k in range(input_ids.shape[0]):
+        #     timestamp_logprob = logprobs[k, self.timestamp_begin :].logsumexp(dim=-1)
+        #     max_text_token_logprob = logprobs[k, : self.timestamp_begin].max()
+        #     if timestamp_logprob > max_text_token_logprob:
+        #         scores[k, : self.timestamp_begin] = -float("inf")
+
+        logprobs = tf.nn.log_softmax(tf.cast(scores, dtype=tf.float32), axis=-1)
```

Comment:
consider using our [`stable_softmax`](https://github.com/huggingface/transformers/blob/42b60f8b02941b0c40c42e150a101eb372c3856e/src/transformers/tf_utils.py#L49), XLA as some known problems with softmax (see the docstring there)